pkg/minishift/docker/image/image_handler.go

```diff
@@ -0,0 +1,348 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/minishift/minishift/pkg/minikube/sshutil"
+	"github.com/minishift/minishift/pkg/util/filehelper"
+	"golang.org/x/crypto/ssh"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// ImageHandler is responsible for the import and export of images into the Docker daemon of the VM
+type ImageHandler interface {
+	// Imports cached images from the host into the Docker daemon of the VM.
+	ImportImages(config *ImageCacheConfig) error
+
+	// Exports the images specified as part of the ImageCacheConfig from the VM to the host.
+	ExportImages(config *ImageCacheConfig) error
+
+	// AreImagesCached returns true if all images specified in the config are cached, false otherwise.
+	AreImagesCached(config *ImageCacheConfig) bool
+}
+
+type DockerImageHandler struct {
+	driver drivers.Driver
+}
+
+func NewDockerImageHandler(driver drivers.Driver) (*DockerImageHandler, error) {
+	return &DockerImageHandler{driver: driver}, nil
+}
+
+func (handler *DockerImageHandler) ImportImages(config *ImageCacheConfig) error {
+	out := handler.getOutputWriter(config)
+
+	files, _ := ioutil.ReadDir(config.HostCacheDir)
+	if len(files) > 0 {
+		fmt.Fprintln(out, "-- Importing cached images ...")
```

Comment:
can we also look soon what to do about the info logging to the console/stdout? In several places we "-- ", or "   " as prefix.